velox/exec/SortBuffer.cpp

```diff
@@ -43,20 +43,28 @@ SortBuffer::SortBuffer(
 
   std::vector<TypePtr> sortedColumnTypes;
   std::vector<TypePtr> nonSortedColumnTypes;
-  std::vector<std::string> sortedSpillColumnNames;
-  std::vector<TypePtr> sortedSpillColumnTypes;
   sortedColumnTypes.reserve(sortColumnIndices.size());
   nonSortedColumnTypes.reserve(input->size() - sortColumnIndices.size());
-  sortedSpillColumnNames.reserve(input->size());
-  sortedSpillColumnTypes.reserve(input->size());
+
+  // Only initialize spill-related variables when enable spill
+  std::vector<std::string> sortedSpillColumnNames;
```

Comment:
Isn't code a bit simpler with less if-else condition except this turns out to be performance issue? This change adds three conditions which are error prone. We could fix the coding style and the one I commented below. Thanks!

Reply:
Indeed, introducing three 'if' conditions adds complexity, which could lead to potential oversights in future coding. However, it does clearly mark that these logics are only related to spill.

Currently, when we disable spill, we still need to incur some (small) cost for initializing spillerStoreType_ (for example, a part of memory on the heap, and a string copy).

Regarding the code style unification, changing input to input_, I will submit a new commit with the update as soon as possible.

Thank you for the review.